ui/src/kapacitor/containers/KapacitorPage.js

```diff
@@ -71,6 +71,23 @@ class KapacitorPage extends Component {
     const {addFlashMessage, source, params, router} = this.props
     const {kapacitor} = this.state
 
+    let doesKapacitorNameAlreadyExist = false
+
+    if (source && source.kapacitors) {
+      source.kapacitors.map(existingKapacitor => {
+        if (existingKapacitor.name === kapacitor.name) {
+          doesKapacitorNameAlreadyExist = true
+        }
+      })
+    }
+    if (doesKapacitorNameAlreadyExist === true) {
+      addFlashMessage({
+        type: 'error',
+        text: `There is already a Kapacitor configuration named "${kapacitor.name}"`,
+      })
+      return
+    }
+
```

Comment:
Another option is to extract the names from each kapacitor object, then ask the result if it includes a known value:
```js
let kapNames = []
if (source && source.kapacitors) {
  kapNames = source.kapacitors.map(k => k.name)
}
if (kapNames.includes(kapacitor.name)) {
```